### PR TITLE
fix(tests): resolve duplicate_service vectorized test suite mismatch and test pollution

### PR DESCRIPTION
**Pull request by harshitanagpal05**
## Summary
We have successfully resolved the test suite collection and runtime failures in the duplicate checking service tests.

### Proposed Changes
- **Unloaded conftest.py Stub Module**: Dynamically removes the `backend.services.duplicate_service` stub injected by `conftest.py` from `sys.modules` at test startup.
- **Implemented Global and Local PyTorch Mocking**: Installed a global mock for the `torch` library to allow tests to call `torch.stack()` and `torch.max()` without failing on dummy `MagicMock` embeddings.
- **Prevented Test Pollution**: Added a `teardown_method` inside `TestCheckDuplicate` to automatically reset the shared `_mock_st.util.cos_sim` mock side-effects and return values between test executions.
- **Corrected Multiple-Ticket Mocks to Vectorized Format**: Refactored `test_picks_best_match_among_multiple_tickets` to model a single vectorized comparison call instead of looping individual comparisons.
closes #790 
### Verification Results
All 24 unit tests now compile and pass cleanly in 14.48s:
```text
backend\tests\test_duplicate_service.py ........................         [100%]
============================= 24 passed in 14.48s =============================

### CODE REVIEW

**Comment by riteshbonthalakoti** [comment]
Superb implementation, @harshitanagpal05! I've successfully resolved all conflicts in your PR and queued it for merging into `gssoc`.

⚠️ **MANDATORY STEPS FOR LEADERBOARD CREDITS**:
To ensure you receive full points, please make sure you have taken 10 seconds to:
- Give us a Star ⭐: https://github.com/ritesh-1918/HELPDESK.AI (Mandatory)
- Follow my GitHub profile 👤: https://github.com/ritesh-1918 (Mandatory)
- Let's connect on LinkedIn 💼: https://www.linkedin.com/in/ritesh1908/ (Mandatory)

Keep up the outstanding work! Let's build together! 🔥

**Comment by riteshbonthalakoti** [comment]
Hi @harshitanagpal05! 🙌

Thank you so much for your excellent contribution: **"fix(tests): resolve duplicate_service vectorized test suite mismatch and test pollution"**! We really appreciate the high-quality code and effort you have put into the platform.

Just a quick, friendly heads-up as we prepare our manual merging and verification queues—please make sure to complete all the mandatory community steps listed below. 

Once those manual steps are verified, we'll get your PR officially merged into the `gssoc` branch (or keep it neatly cataloged if closed as integrated) and assign it the highest possible GSSoC S-Tier labels to maximize your leaderboard points!

Let's build something amazing together! 🚀🔥


---

### 🌟 Community Support & Network Steps (Take 10 Seconds!)
As we prepare our manual verification and merging queues, please make sure you have taken a moment to complete these required steps to finalize your points:
1. ⭐ **Star this repository**: https://github.com/ritesh-1918/HELPDESK.AI (Mandatory)
2. 🍴 **Fork this repository**: https://github.com/ritesh-1918/HELPDESK.AI/fork (Mandatory)
3. 👤 **Follow @ritesh-1918 on GitHub**: https://github.com/ritesh-1918 (Mandatory - manual step)
4. 💼 **Connect on LinkedIn**: https://www.linkedin.com/in/ritesh1908/ (Mandatory)

*Note: Having these steps completed manually is required before your PR points are officially cleared.*

**Comment by riteshbonthalakoti** [comment]
Hi @harshitanagpal05! 🙌

Thank you so much for your excellent contribution: **"fix(tests): resolve duplicate_service vectorized test suite mismatch and test pollution"**! We really appreciate the high-quality code and effort you have put into the platform.

Just a quick, friendly heads-up as we prepare our manual merging and verification queues—please make sure to complete all the mandatory community steps listed below. 

⭐ **Leaderboard tip**: You are already following our project admin @ritesh-1918 and fully registered in our developer network! Your S-Tier point credentials are fully cleared for manual approval! 🙌

Once those manual steps are verified, we'll get your PR officially merged into the `gssoc` branch (or keep it neatly cataloged if closed as integrated) and assign it the highest possible GSSoC S-Tier labels to maximize your leaderboard points!

Let's build something amazing together! 🚀🔥


---

### 🌟 Project Support & Developer Network (Show Some Love!)
As we prepare our manual verification and merging queues, please take a quick moment to ensure you have completed all four community steps:
1. ⭐ **Star this repository**: Helps our AI helpdesk get noticed! [Star the repo here](https://github.com/ritesh-1918/HELPDESK.AI)
2. 🍴 **Fork this repository**: Keep a copy to build your own cool tools! [Fork here](https://github.com/ritesh-1918/HELPDESK.AI/fork)
3. 👤 **Follow @ritesh-1918 on GitHub**: Stay updated on real-time open-source projects! [Follow ritesh-1918 here](https://github.com/ritesh-1918)
4. 💼 **Connect on LinkedIn**: Let's build a strong engineering connection! [Connect with Ritesh on LinkedIn](https://www.linkedin.com/in/ritesh1908/)

*Note: Having all four steps completed manually is required before your PR points are officially cleared.*

**Comment by riteshbonthalakoti** [comment]
Hi @harshitanagpal05! Thanks for the contribution. I have triaged your PR and set it to merge into the `gssoc` branch.

⚠️ **MANDATORY GSSOC ONBOARDING STEPS**:
Before your PR points are finalized on the leaderboard, you **MUST** complete these required steps:
1. ⭐ **Star this repository**: https://github.com/ritesh-1918/HELPDESK.AI (Mandatory)
2. 👤 **Follow the Project Admin**: https://github.com/ritesh-1918 (Mandatory)
3. 💼 **Connect on LinkedIn**: https://www.linkedin.com/in/ritesh1908/ (Mandatory)

Welcome to the HELPDESK.AI developer family! 🚀💻
